modulemd/v2/modulemd-module-stream.c

```diff
@@ -297,6 +297,62 @@ modulemd_module_stream_finalize (GObject *object)
 }
 
 
+static gboolean
+modulemd_module_stream_default_equals (ModulemdModuleStream *self_1,
+                                       ModulemdModuleStream *self_2)
+{
+  if (modulemd_module_stream_get_version (self_1) !=
+      modulemd_module_stream_get_version (self_2))
+    return FALSE;
+
+  if (g_strcmp0 (modulemd_module_stream_get_module_name (self_1),
+                 modulemd_module_stream_get_module_name (self_2)) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (modulemd_module_stream_get_stream_name (self_1),
+                 modulemd_module_stream_get_stream_name (self_2)) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (modulemd_module_stream_get_context (self_1),
+                 modulemd_module_stream_get_context (self_2)) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (modulemd_module_stream_get_arch (self_1),
+                 modulemd_module_stream_get_arch (self_2)) != 0)
+    return FALSE;
+
+  /*need a function for this
+  if(!modulemd_translation_equals (modulemd_module_stream_get_translation (self_1), 
```

Comment:
Skip translation equivalence. It's safe to do because setting the translation is an internal-only function that is handled automatically when adding a module to an index. We only need to know that the untranslated module is equal in this comparison.

Reply:
For additional information: This is really just a back-reference to the translation object stored in the ModulemdModule object this stream belongs to. So as long as the module name is the same, this object is guaranteed to just be a pointer to that translation object, and thus equal.